packages/swapper/src/swappers/ButterSwap/swapperApi/getTradeQuote.ts

```diff
@@ -0,0 +1,232 @@
+
+  // Call ButterSwap /route API
+  const routeResult = await getRoute(
+    butterSwapFromChainId,
+    sellAssetAddress,
+    butterSwapToChainId,
+    buyAssetAddress,
+    bn(amount).shiftedBy(-sellAsset.precision).toString(), // convert to human units
+    slippage,
+  )
+
+  if (routeResult.isErr()) return Err(routeResult.unwrapErr())
+  const routeResponse = routeResult.unwrap()
+  if (!isRouteSuccess(routeResponse)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] ${routeResponse.message}`,
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+
+  const route = routeResponse.data[0]
+  if (!route) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getTradeQuote] No route found',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  // Call ButterSwap /swap API to get calldata and contract info
+  const buildTxResult = await getBuildTx(
+    route.hash,
+    slippage,
+    sendAddress, // from (source chain address)
+    receiveAddress, // receiver (destination chain address)
+  )
+
+  if (buildTxResult.isErr()) return Err(buildTxResult.unwrapErr())
+  const buildTxResponse = buildTxResult.unwrap()
+  if (!isBuildTxSuccess(buildTxResponse)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] /swap failed: ${buildTxResponse.message}`,
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+  const buildTx = buildTxResponse.data[0]
+  if (!buildTx) {
+    return Err(
+      makeSwapErrorRight({
+        message: '[getTradeQuote] No buildTx data returned',
+        code: TradeQuoteError.QueryFailed,
+      }),
+    )
+  }
+
+  // Fee asset for network/protocol fees
+  const feeAsset = _deps.assetsById[feeAssetId]
+  if (!feeAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getTradeQuote] Fee asset not found for chainId ${sellAsset.chainId}`,
+        code: TradeQuoteError.UnsupportedChain,
+      }),
+    )
+  }
+
+  // Map gasFee.amount to networkFeeCryptoBaseUnit using fee asset precision
+  const networkFeeCryptoBaseUnit = bnOrZero(route.gasFee?.amount).gt(0)
+    ? toBaseUnit(route.gasFee.amount, feeAsset.precision)
+    : '0'
+
+  const nativeFeeBaseUnit = bnOrZero(route.swapFee.nativeFee).gt(0)
+    ? toBaseUnit(route.swapFee.nativeFee, feeAsset.precision)
+    : '0'
+  const tokenFeeBaseUnit = bnOrZero(route.swapFee.tokenFee).gt(0)
+    ? toBaseUnit(route.swapFee.tokenFee, sellAsset.precision)
+    : '0'
+
+  let protocolFees: Record<string, ProtocolFee> | undefined = undefined
+  const hasNativeFee = bnOrZero(nativeFeeBaseUnit).gt(0)
+  const hasTokenFee = bnOrZero(tokenFeeBaseUnit).gt(0)
+
+  if (hasNativeFee && hasTokenFee && feeAssetId === sellAsset.assetId) {
+    // If both fees are for the same asset, sum them
+    protocolFees = {
+      [feeAssetId]: {
+        amountCryptoBaseUnit: bnOrZero(nativeFeeBaseUnit).plus(tokenFeeBaseUnit).toString(),
+        requiresBalance: true,
```

Comment:
confirming this should be paid *on top* of the Tx and not a downside?

Reply:
protocolFees is now undefined, so i think this is defunct